core/src/main/java/com/linecorp/armeria/common/HttpResponse.java

```diff
@@ -172,11 +173,7 @@ static HttpResponse from(CompletionStage<? extends HttpResponse> stage) {
         final DeferredHttpResponse res = new DeferredHttpResponse();
         stage.whenComplete((delegate, thrown) -> {
             if (thrown != null) {
-                if (thrown instanceof CompletionException && thrown.getCause() != null) {
-                    res.close(thrown.getCause());
-                } else {
-                    res.close(thrown);
-                }
+                res.close(Throwables.getRootCause(thrown));
```

Comment:
`getRootCause()` can return `null` and you need to handle it.

Reply:
I think it's implementation will return self if cause is null.

Reply:
Sorry, I misread the code. :-)